README.md
# Hadleyville Online

An web version of my rules light Old West RPG, [Hadleyville](https://madinkbeard.itch.io/hadleyville) (free at itch.io).

I have plans to add more features as time permits, see the issues in github.

~~This is coded in Angular because I need to learn it for work, so there might be problems with how I use Angular.~~ Refactored to not use a framework.

This is also me working out the issues with my [RPG Table Randomizer library](https://github.com/derikb/rpg-table-randomizer).

## Install

To use, just point your server at `/dist` or locally, run `npm i && npm run start` to install dependencies and start a dev server that is viewable in your browser.

For development:

- `npm install` to get the dependencies.
- `npm run start` will start the esbuild dev server so you can view in your browser.
- `npm run build` will generate the build files for `/dist` (just js and css).


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

## [2.2.0](https://github.com/derikb/hadleyville-online/compare/v2.1.1...v2.2.0) (2022-03-12)


### Features

* **factions:** Add basic faction model/service/display ([c89ab89](https://github.com/derikb/hadleyville-online/commit/c89ab89919a43e2ffea0c63501b9d733d655a8cc)), closes [#17](https://github.com/derikb/hadleyville-online/issues/17)
* **factions:** Relationships based on reaction roll ([f4ba745](https://github.com/derikb/hadleyville-online/commit/f4ba745b0a620b2637c286c2710127c80bb934d7)), closes [#17](https://github.com/derikb/hadleyville-online/issues/17)
* **factions:** Show factions on map ([6022281](https://github.com/derikb/hadleyville-online/commit/6022281c0ae897ab9b186cb611f93d348961b736)), closes [#17](https://github.com/derikb/hadleyville-online/issues/17)
* **links:** More links from notes to npc/faction objects ([fe0a253](https://github.com/derikb/hadleyville-online/commit/fe0a253761dabcb75e79a020780046a8a7d0c8d9)), closes [#25](https://github.com/derikb/hadleyville-online/issues/25)
* **map:** Add pcs to map ([7d02157](https://github.com/derikb/hadleyville-online/commit/7d02157fe9bae823d9da94f85d968f87b08691db))
* **notes:** Add links to notes ([fb869f9](https://github.com/derikb/hadleyville-online/commit/fb869f93cb781b62eeaacfc14c59ec0f3babb4e8)), closes [#25](https://github.com/derikb/hadleyville-online/issues/25)
* **pc:** Add relationships to pcs ([d4de8ae](https://github.com/derikb/hadleyville-online/commit/d4de8ae4d2e3df82cbe678dcff867998a4c34737)), closes [#16](https://github.com/derikb/hadleyville-online/issues/16)
* **pc:** Added tables of example traits/skills ([41ecb9f](https://github.com/derikb/hadleyville-online/commit/41ecb9f91fcaa3fc4f3482771eadcc5506dc59f0)), closes [#18](https://github.com/derikb/hadleyville-online/issues/18)
* **relationships:** Improve table, add roll to form ([bd61f09](https://github.com/derikb/hadleyville-online/commit/bd61f0955970e77ac2272c545651f871aaff186e)), closes [#22](https://github.com/derikb/hadleyville-online/issues/22)
* **tables:** Add oneof token to tables ([db2b53a](https://github.com/derikb/hadleyville-online/commit/db2b53a624635ae68aa2df53a233014eaa2c4391))


### Bug Fixes

* **tables:** Fix table result modal rerolls ([9b462a4](https://github.com/derikb/hadleyville-online/commit/9b462a466338c00a60636ef64c52d0b813babcf2)), closes [#23](https://github.com/derikb/hadleyville-online/issues/23)

### [2.1.1](https://github.com/derikb/hadleyville-online/compare/v2.1.0...v2.1.1) (2022-02-06)


### Bug Fixes

* **map modal:** Move into route output ([3132800](https://github.com/derikb/hadleyville-online/commit/3132800e21cc020d5ed7ec87479137f24c94acf8)), closes [#20](https://github.com/derikb/hadleyville-online/issues/20)
* **route:** Fix for empty output ([bafb7e7](https://github.com/derikb/hadleyville-online/commit/bafb7e78cab1d2a21dbc159ce9fb5bce01bb1cfa)), closes [#19](https://github.com/derikb/hadleyville-online/issues/19)

## [2.1.0](https://github.com/derikb/hadleyville-online/compare/v2.0.0...v2.1.0) (2022-02-06)


### Features

* **characters:** Add character page ([f32ee49](https://github.com/derikb/hadleyville-online/commit/f32ee495e542cda2055d84333b63a449bf79b318)), closes [#1](https://github.com/derikb/hadleyville-online/issues/1)
* **map:** Account for viewport being smaller than coords ([cff7fe5](https://github.com/derikb/hadleyville-online/commit/cff7fe59993f911ca75579fe4cc408599751cc92))
* **map:** Add arrow on link label ([1a52ef6](https://github.com/derikb/hadleyville-online/commit/1a52ef6b86997fcf62e5f641e7851fc86bebd9fa))
* **map:** Show two way relationships ([5e42926](https://github.com/derikb/hadleyville-online/commit/5e4292670e0fe359d8415b10fc9e56fa7dbb5df7))
* **npcs:** Add relationships for NPCs ([d41143e](https://github.com/derikb/hadleyville-online/commit/d41143e7fcf761e6d0dc0ba82d28311acb63ae0c))
* **relationships:** Basic map for NPCs ([a86afc8](https://github.com/derikb/hadleyville-online/commit/a86afc8edd2c5cfbb4a516c5a1fd113c93ede28f)), closes [#3](https://github.com/derikb/hadleyville-online/issues/3)
* **routing:** Add routing for pages ([9657971](https://github.com/derikb/hadleyville-online/commit/96579713ee47f826387f8f14347729c097a1ebc0)), closes [#15](https://github.com/derikb/hadleyville-online/issues/15)
* **tables:** Add reroll to results modal ([b7d021e](https://github.com/derikb/hadleyville-online/commit/b7d021ef357778c4fcaa1f5cabfbc44c348c2d2f)), closes [#14](https://github.com/derikb/hadleyville-online/issues/14)


### Bug Fixes

* **css:** Fix dupe id causing weird styling in form ([858d91e](https://github.com/derikb/hadleyville-online/commit/858d91eaaadfd7c826c598e30d55d2cd8f90de9d))
* **modals:** Scroll content when overflowing ([1d08b57](https://github.com/derikb/hadleyville-online/commit/1d08b5745018d0ec87d0f39a08ac70197b6ed054))

## [2.0.0](https://github.com/derikb/hadleyville-online/compare/v1.0.0...v2.0.0) (2022-01-13)


### Features

* Focus management improvements ([8156b0e](https://github.com/derikb/hadleyville-online/commit/8156b0eba1826eb9ead4c3e5d04f7f33dbeb3f36))
* New table items, new rules ([817dbcd](https://github.com/derikb/hadleyville-online/commit/817dbcdfd0e57b5106f97c0d7dd7008ec1817ad2))

src/data/tables.js
export default [
    {
        id: 3,
        key: 'npcs',
        title: 'NPC Characteristics',
        sequence: ['size', 'manner', 'physical_trait', 'quirk', 'speech'],
        tables: {
            size: [
                'Short',
                'Thin',
                'Average',
                'Tall',
                'Large',
                'Pick Two'
            ],
            dress: [
                'Cleancut',
                'Filthy',
                'Fancy',
                'Practical',
                'Rich',
                'Uniform'
            ],
            manner: [
                'Gregarious',
                'Icy',
                'Pious',
                'Flirtacious / Obsequious',
                'Aggressive / Critical',
                'Reserved'
            ],
            physical_trait: [
                'Missing Limb',
                'Hairy',
                'Unusual hair',
                'Muscular',
                'Unusually Attractive',
                'Unusually Ugly'
            ],
            quirk: [
                'Touchy',
                'Fingers Talisman ({{oneof:Crucifix|Ring|Gun|Cards}})',
                'Ticket/Fidgety',
                'Bad Sense of {{oneof:Sight|Hearing}}',
                '{{oneof:Drinks|Smokes|Chews}} Profusely',
                'Gestures a lot'
            ],
            speech: [
                'Stutters',
                'Florid',
                'Terse',
                'Curses excessively',
                'Laughs a lot',
                'Meandering'
            ]
        }
    },
    {
        id: 4,
        key: 'jobs',
        title: 'NPC Jobs',
        display_opt: [
            { table: 'worker', hide_table: true },
            { table: 'merchant', hide_table: true },
            { table: 'craftsman', hide_table: true },
            { table: 'professional', hide_table: true },
            { table: 'wealthy', hide_table: true },
            { table: 'neer_do_well', hide_table: true }
        ],
        tables: {
            default: [
                { label: 'Worker', subtable: 'worker', print: false },
                { label: 'Merchant', subtable: 'merchant', print: false },
                { label: 'Craftsman', subtable: 'craftsman', print: false },
                { label: 'Professional', subtable: 'professional', print: false },
                { label: 'Wealthy', subtable: 'wealthy', print: false },
                { label: 'Ne\'er Do Well', subtable: 'neer_do_well', print: false }
            ],
            worker: [
                'Farmer',
                'Bartender',
                'Clerk',
                'Cowboy',
                'Soldier',
                'Laundress',
                'Prostitute',
                'Logger',
                'Parent',
                'Miner',
                'Trapper',
                'Unemployed'
            ],
            merchant: [
                'General Store',
                'Bathhouse',
                'Hotel/Boarding',
                'Saloon',
                'Pimp',
                'Hardware',
                'Livery',
                'Foreman/Manager',
                'Food/Cafe',
                'Freight',
                'Haberdasher',
                'Feed Barn'
            ],
            craftsman: [
                'Blacksmith',
                'Wainwright',
                'Brewer/Distiller',
                'Saddler',
                'Tailor',
                'Barber',
                'Cook',
                'Butcher',
                'Gunsmith',
                'Sawmill',
                'Carpenter',
                '?'
            ],
            professional: [
                'Doctor',
                'Lawyer',
                'Newspaper Editor',
                'Teacher',
                'Sheriff',
                'Army Officer',
                'Preacher',
                'Telegraph Operator',
                'Conductor',
                'Assayer',
                'Surveyor',
                'Marshal'
            ],
            wealthy: [
                'Mine Owner',
                'Cattle Baron',
                'Trust Fund Child',
                'Banker',
                'Politician',
                'Nouveau Riche'
            ],
            neer_do_well: [
                'Gambler',
                'Addict',
                'Gunfighter',
                'Actor',
                'Musician',
                'Dancer',
                'Thug',
                'Gossip',
                'Thief',
                'Gigolo',
                'Hermit',
                'Vigilante'
            ]
        }
    },
    {
        id: 5,
        key: 'goals',
        title: 'NPC Goals',
        display_opt: [
            { table: 'long_term', hide_table: true },
            { table: 'short_term', hide_table: true },
            { table: 'secrets', hide_table: true }
        ],
        tables: {
            long_term: [
                'Wealth',
                'Safety of {{oneof:Family|Self}}',
                'Civic Good',
                'Knowledge/Skill/Craft',
                'Piety/Charity',
                'Status/Power',
                'Peace & Quiet',
                'Passion'
            ],
            short_term: [
                'Business Deal',
                'Pleasure/Addiction',
                'Property',
                'Specific Object',
                'Money',
                'Job/Work',
                'Love/Lust/Desire',
                'Revenge/Hate'
            ],
            secrets: [
                'Love’s another’s spouse ({{oneof:mutual|unrequited|unexpressed}})',
                'Fled a murder rap ({{oneof:innocent|guilty}})',
                'Fraud at profession',
                'Hidden vice ({{oneof:dope|alcohol|sex}})',
                'Secret child in town',
                'Ran with a bad bunch of hombres',
                'Owes a lot of money / Broke',
                'Hallucinatory visions ({{oneof:religious|psychotic}})'
            ]
        }
    },
    {
        id: 6,
        key: 'relationships',
        title: 'NPC Relationships',
        display_opt: [
            { table: 'default', hide_table: true },
            { table: 'specific', hide_table: true },
            { table: 'faction', hide_table: false }
        ],
        tables: {
            default: [
                { label: 'Hate', weight: 1 },
                { label: 'Dislike', weight: 5 },
                { label: 'Suspicious', weight: 7 },
                { label: 'Neutral', weight: 10 },
                { label: 'Like', weight: 7 },
                { label: 'Friend', weight: 5 },
                { label: 'Love', weight: 1 }
            ],
            specific: [
                'Friends {{oneof:best|former|}}',
                'Lovers ({{oneof:current|former|unrequited}})',
                'Enemies ({{oneof:long term|recent|former}})',
                'Rivalry ({{oneof:business|love}})',
                'Loaner/Debtor',
                'Comrades ({{oneof:current|former}}) ({{oneof:business|army|mines|banditry|cowboys}})',
                'Employer/Employee ({{oneof:current|former}})',
                'Family ({{oneof:happy|secret|estranged}})'
            ],
            faction: [
                'Community leaders (business owners)',
                'Pious church goers',
                'Miners/Loggers {{oneof:union|}}',
                'Ranchers',
                'The Law (sheriff, deputies, judges, posse members, etc.)',
                'Homesteaders',
                'Bandits'
            ]
        }
    },
    {
        id: 7,
        key: 'events',
        title: 'Events',
        tables: {
            regular: [
                'Wedding ({{oneof:jealousy|drunken revelry}})',
                'Funeral ({{oneof:murder|inheritance}})',
                'Market day {{oneof:theft|rivalry|strangers}}',
                'Holiday',
                'Arrival of ({{oneof:Stage|Train}})',
                'Sunday services',
                'Payday at the {{oneof:Mine|Ranch|Camp}}'
            ],
            crimes: [
                'Drunken brawl',
                'Murder',
                'Theft',
                'Kidnapping',
                'Bandit raid',
                'Shootout',
                '({{oneof:Stage|Train}}) ambushed',
                'Arson'
            ],
            natural: [
                'Resource Depletion',
                'Sickness',
                'Rainstorm',
                'Windstorm',
                'Fire',
                'Wild animals'
            ],
            other: [
                'Territorial dispute ({{oneof:ranchers|farmers|both}})',
                'Heartless capitalist tries to run things',
                'A dark past catches up with someone',
                'Mine collapse/Logging accident',
                'Election',
                'Workers strike',
                'Lynch mob',
                'New business opens',
                'Square dance'
            ]
        }
    },
    {
        id: 8,
        key: 'stranger',
        title: 'A Stranger Comes to Town',
        tables: {
            default: [
                'Gunslinger',
                'Pinkerton Detective',
                'Itinerant Preacher',
                '{{oneof:Son|Father|Daughter|Lover}} thought to be dead',
                'Family member with a {{oneof:problem|past|secret}}',
                'Ex-con {{oneof:stranger|relative|old friend}}',
                'Snake Oil Salesman',
                'Rabblerouser',
                'Swindler {{oneof:man|woman|couple|family}}',
                'Rich ({{oneof:worldly|naive}}) {{oneof:man|couple|woman}} from back East',
                'Bandit {{oneof:with|without}} his gang',
                'Family looking to settle',
                'Destitute men looking for work',
                'Former soldiers ({{oneof:hungry|crazed|in pursuit|pursued}})',
                'Bounty hunter with a {{oneof:recent|old}} warrant for {{oneof:townsperson|bandit|gunslinger}}',
                'Traveling salesman',
                '{{oneof:Troublesome|Pitiful}} {{oneof:Man|Woman}} comes to get {{oneof:Family|Friend}} out of jail',
                'Politician ({{oneof:campaigning|canvassing}})',
                'Woman fleeing {{oneof:fiance|husband}}',
                'Travelling Troupe of Actors'
            ]
        }
    },
    {
        id: 10,
        key: 'items',
        title: 'Items',
        tables: {
            default: [
                { label: 'weapons', subtable: 'weapons', print: true },
                { label: 'valuables', subtable: 'valuables', print: true },
                { label: 'clothes', subtable: 'clothes', print: true },
                { label: 'consumables', subtable: 'consumables', print: true },
                { label: 'papers', subtable: 'papers', print: true },
                { label: 'other', subtable: 'other', print: true }
            ],
            weapons: [
                'Famous gunslinger’s pistol',
                'Knife',
                'Gun ({{oneof:shotgun|pistol|rifle}})',
                'Bandolier of bullets',
                'Cavalry Sword',
                'Ax',
                'Whip'
            ],
            valuables: [
                'Wedding band of {{oneof:live|dead}} spouse',
                'Thoroughbred horse',
                'Unopened bottle of imported liquor',
                'Wallet of bills',
                'Pouch of gold',
                'Combination to safe on a photo',
                'Crucifix'
            ],
            clothes: [
                'Hat {{oneof:lady’s|man’s}} {{oneof:fine|plain}}',
                'Boots {{oneof:new|muddy|decorative}}',
                'Dress {{oneof:fancy|torn|bloodied}}',
                'Shirt {{oneof:fancy|bloodstained|torn}}',
                'Gun belt',
                'Earrings {{oneof:plain|bejeweled}}'
            ],
            consumables: [
                'Bottle of whisky',
                'Jerky',
                'Tobacco pouch and rolling papers',
                'Canned peaches',
                'Sweet buns',
                'Wine bottle'
            ],
            papers: [
                'Newspaper {{oneof:local|nearby|far}}',
                'Deed to mines',
                'Arrest warrant',
                'Letter {{oneof:incriminating|sentimental}}',
                'Dime novel',
                'Daguerrotype'
            ],
            other: [
                'Dice [loaded]',
                'Playing cards {{oneof:fresh|used}}',
                'Keys to {{oneof:house|business|lockbox|jail}}',
                'Hammer {{oneof:and nails|}}',
                'Pipe and matches',
                'Lantern'
            ]
        }
    },
    {
        id: 9,
        key: 'town_landscape',
        title: 'Landscape Prominent Feature',
        tables: {
            default: [
                'Town build on the banks of a winding stream crossed by makeshift bridges.',
                'Town build upon rocky outcrops, street(s) wind back and forth, buildings are all at different levels.',
                'Town grew around a single giant tree, the rest of the land is flat and barren.',
                'Town sits in a narrow valley, overlooked by ridges, main road snakes down to town and then back up and out.',
                'Town built at foot of cliff where the first mine was found.',
                'Town expanded from a small waterfall and pool amongst tall evergreens.'
            ]
        }
    },
    {
        id: 11,
        key: 'dressing',
        title: 'Set Dressing',
        display_opt: [
            { table: 'outside', hide_table: true },
            { table: 'inside', hide_table: true }
        ],
        tables: {
            outside: [
                'Balconies',
                'Boardwalks',
                'Hitching posts',
                'Water troughs',
                'Fences',
                'Barrels',
                'Mud Puddles',
                'Hand scrawled signs',
                'Water wagon',
                'Log bridges',
                'Water wheels',
                'Framed out buildings',
                'Tents',
                'Horses and Wagons',
                'Posters (Wanted, Ads, Broadsheets)'
            ],
            inside: [
                'Piano',
                'Billiard table',
                'Dart board',
                'Music box',
                'Spittoons',
                'Bed pans',
                'Paintings {{oneof:landscape|pretty woman|historical figure}}',
                'Mismatched chairs',
                'Makeshift tables',
                'Ladders',
                'Heavy curtains',
                'Wash tubs',
                'Valise filled with dresses',
                'Old Clock',
                'Brightly upholstered chair/couch',
                'Crystal decanter filled with whiskey'
            ]
        }
    },
    {
        id: 1,
        key: 'locations',
        title: 'Locations Outside of Town',
        tables: {
            default: [
                'Train {{oneof:in motion|broken down|stopped by blocked tracks}}',
                'Family Farm {{oneof:abandoned|prospering|failing}}',
                'Cattle Ranch',
                'Mine {{oneof:abandoned|active}}',
                'Grazing land {{oneof:disputed|}}',
                'River {{oneof:rapids|}}',
                'Rocky hills',
                'Deep woods',
                'Religious camp',
                'Bandit camp',
                'Log cabin {{oneof:abandoned|hideout|hermit}}'
            ]
        }
    },
    {
        id: 2,
        key: 'resources',
        title: 'Settlement Resources',
        tables: {
            default: [
                'Gold',
                'Lumber',
                'Farmland',
                'Water',
                'Railway',
                'Furs/Game'
            ]
        }
    },
    {
        id: 12,
        key: 'town_name',
        title: 'Town Names',
        display_opt: [
            { table: 'main', hide_table: true },
            { table: 'ending1', hide_table: true },
            { table: 'ending2', hide_table: true }
        ],
        tables: {
            default: [
                { label: '{{table:this:main}}{{table:this:ending1}}', weight: 4 },
                { label: '{{table:this:main}} {{table:this:ending2}}', weight: 4 },
                { label: 'Fort {{table:this:main}}', weight: 1 }
            ],
            main: [
                'Hadley',
                'Smith',
                'Adams',
                'Snake',
                'Elk',
                'Fire',
                'Coal',
                'Oak',
                'Maple',
                'Elm',
                'Wilder',
                'Jackson',
                'Washington',
                'Revere',
                'Franklin',
                'Shallow',
                'Rapid',
                'Gold',
                'Summer',
                'Spring',
                'Nugget',
                'Mary'
            ],
            ending1: [
                'ville',
                'town',
                'berg',
                'woods'
            ],
            ending2: [
                'Hollow',
                'Hills',
                'City',
                'Falls',
                'Rapids'
            ]
        }
    },
    {
        id: 13,
        key: 'character_traits',
        title: 'Character Traits (examples)',
        display_opt: [
            { table: 'positive', hide_table: true },
            { table: 'negative', hide_table: true }
        ],
        tables: {
            positive: [
                'Smart',
                'Strong',
                'Persuasive',
                'Faithful',
                'Quick',
                'Educated',
                'Empathetic',
                'Cunning',
                'Popular',
                'Experienced',
                'Wise',
                'Animal Friend',
                'Lucky',
                'Attractive',
                'Amusing',
                'Robust Health'
            ],
            negative: [
                'Weak',
                'Stupid',
                'Ignorant',
                'Ugly',
                'Despised',
                'Unlucky',
                'Slow',
                'Filthy',
                'Hot Tempered',
                'Impulsive',
                'Annoying',
                'Unhealthy'
            ]
        }
    },
    {
        id: 14,
        key: 'character_skills',
        title: 'Character Skills (examples)',
        display_opt: [
            { table: 'default', hide_table: true }
        ],
        tables: {
            default: [
                'Riding',
                'Driving',
                'Rustling',
                'Pistols',
                'Rifles',
                'Cards',
                'Writing',
                'Mining',
                'Medicine',
                'Charm',
                'Sales',
                'Farming',
                'Lumberjack',
                'Blacksmith',
                'Hunting',
                'Acting',
                'Music'
            ]
        }
    }
];
